### PR TITLE
Lsn1 content - Introduction and setup

### PR DESCRIPTION
**Pull request by CarlyLanglois**
still needs some Windows notes for visual studio setup.
and potentially a sample of importing a project into vs from github.
as well as maybe more material on the project structure?

### CODE REVIEW

File: src/chapters/introduction-and-setup/why-csharp.rst [hunk 1]
```diff
@@ -0,0 +1,25 @@
+Why Learn C#?
+=============
+
+.. index:: ! C#, csharp
+
+This course is an introduction to **C#** and the .NET framework. C# is a
+programming language developed by Microsoft and used in almost all of its products. 
+This course is designed for learners who are already familiar with at least one
+programming language. Thus, we'll move quickly through the syntax rules and
+procedural basics of the language (ie, how does one write a for loop in C#? a
```
Comment (gildedgardenia) [inline src/chapters/introduction-and-setup/why-csharp.rst]: Syntax highlighting on "for"

File: src/chapters/introduction-and-setup/why-csharp.rst [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+it's wise to start learning new ones early on in your career. 
+
+.. index:: Visual Studio IDE, .NET 
+
+Beyond the basics of programming in C#, you'll learn some key *object oriented*
+concepts to enhance the capabilities of your programs. You'll also download a 
+program called **Visual Studio IDE** to get familiar with the tools of the C# developer.
+In the final lessons of this course, you will use a C# framework called **.NET**.
+This framework gives us the scaffolding necessary to create *MVC* programs, another
+concept we'll cover, effectively and efficiently.
```
Comment (gildedgardenia) [inline src/chapters/introduction-and-setup/why-csharp.rst]: The way this sentence ends seems a little awkward. I would replace the commas with parentheses to move "another concept we'll cover" more to the side.

File: src/chapters/introduction-and-setup/setup.rst [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+.. index:: ! integrated development environment, IDE
+
+.. _install-visual-studio:
+
+Visual Studio
+-------------
+
+Visual Studio is an **integrated development environment (IDE)**. An IDE is like a text
+editor on steroids. It not only allows you to write and edit code, but also contains many 
+features that enhance the coding experience. Visual Studio offers
```
Comment (gildedgardenia) [inline src/chapters/introduction-and-setup/setup.rst]: Maybe rephrase to "We can write and edit code *and* make use of additional features that enhance our coding experience."

File: src/chapters/introduction-and-setup/setup.rst [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+code completion hints, debugging, and even it's own compiler. We'll be using it throughout
+this course, so it's time to get familiar with some of the basics.
+
+.. admonition:: Note
+
+   Visual Studio IDE is not the same application as the text editor Visual Studio Code.
```
Comment (gildedgardenia) [inline src/chapters/introduction-and-setup/setup.rst]: "... editor, Visual Studio Code."

File: src/chapters/introduction-and-setup/setup.rst [hunk 5]
```diff
@@ -0,0 +1,88 @@
...
+
+.. index:: ! .NET Core, ! Software Development Kit, ! SDK
+
+.NET Core
+---------
+
+**.NET Core** is a set of tools for developing software in a number of different programming languages, including C#.
+The tools together make what is known as a **software development kit**, or **SDK**.
+Unlike it's predecessor, just called .NET or .NET Framework, .NET Core is open-source and available for development on several 
+operating systems. The SDK provides the runtime environment, as well as the virtual machine for compiling
```
Comment (gildedgardenia) [inline src/chapters/introduction-and-setup/setup.rst]: Maybe rephrase to "open-source and available on several operating systems." The "for development" seems unnecessary.

File: src/chapters/introduction-and-setup/setup.rst [hunk 6]
```diff
@@ -0,0 +1,88 @@
...
+.. index:: ! .NET Core, ! Software Development Kit, ! SDK
+
+.NET Core
+---------
+
+**.NET Core** is a set of tools for developing software in a number of different programming languages, including C#.
+The tools together make what is known as a **software development kit**, or **SDK**.
+Unlike it's predecessor, just called .NET or .NET Framework, .NET Core is open-source and available for development on several 
+operating systems. The SDK provides the runtime environment, as well as the virtual machine for compiling 
+and running C# programs.
```
Comment (gildedgardenia) [inline src/chapters/introduction-and-setup/setup.rst]: Maybe rephrase to "the runtime environment and the virtual machine..."

File: src/chapters/introduction-and-setup/setup.rst [hunk 7]
```diff
@@ -0,0 +1,88 @@
...
+.NET Core
+---------
+
+**.NET Core** is a set of tools for developing software in a number of different programming languages, including C#.
+The tools together make what is known as a **software development kit**, or **SDK**.
+Unlike it's predecessor, just called .NET or .NET Framework, .NET Core is open-source and available for development on several 
+operating systems. The SDK provides the runtime environment, as well as the virtual machine for compiling 
+and running C# programs. 
+.NET Core also notably contains the class library. This is the built-in code that takes care of common programming items
+like date object formatting, for example.
```
Comment (gildedgardenia) [inline src/chapters/introduction-and-setup/setup.rst]: "common programming items, such as date object formatting."

File: src/chapters/introduction-and-setup/setup.rst [hunk 8]
```diff
@@ -0,0 +1,88 @@
...
+like date object formatting, for example. 
+
+.. _compiling-csharp:
+
+A step-by-step walk-through of the process:
+
+#. We write code in C#.
+#. The source code is compiled (like translating) into another intermediate language.
+#. The intermediate code is read by a runtime program included in the .NET SDK.
+#. The runtime environment translates the intermediate code into machine-readable language.
```
Comment (gildedgardenia) [inline src/chapters/introduction-and-setup/setup.rst]: This walkthrough needs a little intro. It felt a little out of left field to me.

File: src/chapters/introduction-and-setup/setup.rst [hunk 9]
```diff
@@ -0,0 +1,88 @@
...
+
+Fortunately for us, .NET Core can be installed along with Visual Studio IDE.
+
+Windows Users vs. Mac Users
+---------------------------
+
+Windows users will setup their C# development environment with a community version of Visual Studio IDE. Mac users
+need to download and install a related, yet different, tool called Visual Studio for Mac.
```
Comment (gildedgardenia) [inline src/chapters/introduction-and-setup/setup.rst]: Instead of "related, yet different", what about "an equivalent" or "a different version of Visual Studio called"?

File: src/chapters/introduction-and-setup/creating-a-csharp-project.rst [hunk 10]
```diff
@@ -0,0 +1,151 @@
...
+      }
+
+   Click on the triangle button to run the project and see the output.
+
+   .. admonition:: Tip
+   
+      The first time you run a console app in Visual Studio, you may be prompted
+      to allow VS to access the terminal. This is ok.
+
+#. A console window should pop up with the line "Hello World"" printed.
```
Comment (gildedgardenia) [inline src/chapters/introduction-and-setup/creating-a-csharp-project.rst]: Double quotation mark

File: src/chapters/introduction-and-setup/creating-a-csharp-project.rst [hunk 11]
```diff
@@ -0,0 +1,151 @@
...
+.. index:: ! solution
+
+Hello, Solution!
+----------------
+
+You've just created your first C# project. Congrats! In fact, you've also just created your own C# **solution**.
+Your ``HelloWorld`` project is nested within a solution called ``HelloWorld``. A solution behaves like a container for
+related projects and other Visual Studio settings. 
+
+A C# project contains all the code to run a particular application. Along with ``Program.cs`` file you ran just a
```
Comment (gildedgardenia) [inline src/chapters/introduction-and-setup/creating-a-csharp-project.rst]: "Along with the ``Program.cs`` file"

**Review by gildedgardenia — CHANGES_REQUESTED** [review]
Looks pretty good! I noted somethings that were more personal preference than anything. I didn't know if the Windows instructions had been verified on a Windows machine, so if they have, let me know and I can promptly re-review!

File: src/chapters/introduction-and-setup/why-csharp.rst [hunk 12]
```diff
@@ -0,0 +1,25 @@
...
+=============
+
+.. index:: ! C#, csharp
+
+This course is an introduction to **C#** and the .NET framework. C# is a
+programming language developed by Microsoft and used in almost all of its products. 
+This course is designed for learners who are already familiar with at least one
+programming language. Thus, we'll move quickly through the syntax rules and
+procedural basics of the language (ie, how does one write a ``for``z loop in C#? a 
+conditional statement?).
```
Comment (gildedgardenia) [inline src/chapters/introduction-and-setup/why-csharp.rst]: This syntax highlighting is not working. Perhaps, the extra "z" is breaking it?

File: src/chapters/introduction-and-setup/why-csharp.rst [hunk 13]
```diff
@@ -0,0 +1,25 @@
+Why Learn C#?
+=============
+
+.. index:: ! C#, csharp
+
+This course is an introduction to **C#** and the .NET framework. C# is a
```
Comment (gildedgardenia) [inline src/chapters/introduction-and-setup/why-csharp.rst:6]: I would make ".NET framework" bold.

File: src/chapters/introduction-and-setup/windows-setup.rst [hunk 14]
```diff
@@ -0,0 +1,87 @@
+Windows Users: Visual Studio Community Edition
+==============================================
+
+#. Download the Community version of Visual Studio IDE from `this page <https://visualstudio.microsoft.com/downloads/?utm_medium=microsoft&utm_source=docs.microsoft.com&utm_campaign=button+cta&utm_content=download+vs2019>`__.
+
+   .. admonition:: Note
+
+      There are options for other software downloads on this page, Visual Studio Code and Visual Studio for Mac.
```
Comment (gildedgardenia) [inline src/chapters/introduction-and-setup/windows-setup.rst]: This sentence reads a little awkwardly. Perhaps, "Users can select from multiple options for software downloads, such as Visual Studio Code and Visual Studio for Mac, on this page. Do NOT select either of these options!"

File: src/chapters/introduction-and-setup/creating-a-csharp-project.rst [hunk 15]
```diff
@@ -0,0 +1,177 @@
...
+
+   You'll see the project file tree containing a file called ``Program.cs`` 
+   in a pane called *Solution Explorer*.
+   
+   .. figure:: figures/vs-windows-new-project.png
+      :alt: WINDOWS: A new Visual Studio Project.
+
+      WINDOWS: A new Visual Studio project.
+
+#. You are new to C# the language and we'll go over the syntax present in ``Program.cs``
```
Comment (gildedgardenia) [inline src/chapters/introduction-and-setup/creating-a-csharp-project.rst]: I would remove "the language"

**Review by gildedgardenia — APPROVED** [review]
I like the setup instructions and only noticed a few minor things. Before you merge, you should review your alt text. I think some of it could be more descriptive since certain defaults are selected or arrows are present pointing to key buttons. Maybe more like "Visual Studio for Mac Installation window with aforementioned menu defaults selected."

File: src/chapters/introduction-and-setup/why-csharp.rst (hunk 13 above)
Comment (CarlyLanglois) [reply src/chapters/introduction-and-setup/why-csharp.rst:6]: this is funny, Jim tells me not to bolden anything unless its getting defined so when i wrote this bit for the java book, he had me remove the entry since we dont define the term until later. I've been trying to follow that rule since
